### DISCUSSION TITLE
Multiple figure/table captions in a single code block

### DISCUSSION THREAD

**Discussion by sutzig** [quarto-dev/quarto-cli#1036] [Q&A] [answered]
Is it possible to define multiple figure captions for multiple tables/figures in one  R code block?

**Comment by jjallaire** [comment]
Yes, see the docs here: https://quarto.org/docs/authoring/figures.html#figure-captions

If you use an array of `fig-cap` as opposed to an array of `fig-subcap` then you can assign a caption to each figure (as opposed to sub-captions)

**Reply by sutzig** [reply]
If I understand correctly, the top of the code chunk should be 

```
{r}

#| fig-cap: Charts
#| fig-subcap: ["Cars", "Pressure"]
#| layout-ncol: 2

```

This, however, throw the following error:

```
Error in if (nzchar(caption)) { : the condition has length > 1
Calls: .main ... sew -> <Anonymous> -> lapply -> FUN -> wrap_asis_output
Execution halted
```

It seems to be that the source of this error is trying to use multiple captions/subcaptions with gt and kable tables.

**Comment by jjallaire** [comment]
Could you provide the full source code for the example that's giving you this error? I tried the following and it works as expected:

````markdown
```{r}
#| label: fig-charts
#| fig-cap: ["Cars", "Pressure"]
#| layout-ncol: 2

plot(cars)
plot(pressure)
```
````

**Comment by sutzig** [comment]
Thank you for you attention in this matter. The example you provided works in my environment as well. It seems that the error is being through when gt or kable tables are involved. Both examples below generate the error
```
Quitting from lines 36-42 (test.rmarkdown) 
Error in if (nzchar(caption)) { : the condition has length > 1
Calls: .main ... sew -> <Anonymous> -> lapply -> FUN -> wrap_asis_output

Execution halted
```
```
```{r}
#| label: fig-charts
#| fig-cap: ["Cars", "Pressure"]
#| layout-ncol: 2

kableExtra::kbl(head(cars))
plot(pressure)
```

```
```{r}
#| label: fig-charts
#| fig-cap: ["Cars", "Pressure"]
#| layout-ncol: 2

kableExtra::kbl(head(cars))
gt::gt(head(pressure))
```
It's worth noting that the following works as expected

```
```{r}
#| label: fig-charts
#| fig-cap: "Cars"

gt::gt(head(cars))
```
And, interestingly, so does this (though, obviously, the same caption is applied to both tables and the label placement/orientation  is not the same as the plot example, but I assume that can be tweaked)
```
```{r}
#| label: fig-charts
#| fig-cap: ["Cars"]
#| layout-ncol: 2

gt::gt(head(cars))
gt::gt(head(pressure))
```

I realize both of these table functions have their own means of providing captions. I was just hoping I might be able to standardize captioning across charts/plots and tables in a single quarto document.

**Comment by jjallaire** [comment]
Okay, I see the problem. For tables you need to use `tbl-cap` rather than `fig-cap`: https://quarto.org/docs/authoring/tables.html#computations

**Reply by sutzig** [reply]
That  works in so far as it doesn't throw an error. It also does not give any captions. I'm also confused by the output. I'm on a Windows machine with R version 4.2.0 (2022-04-22 ucrt) and Quarto 0.9.494. The echo, layout, and (obviously) caption instructions are not being recognize. If I add echo as `{r, echo = FALSE}` it works.
```
```{r}

#| label: tbl-example
#| tbl-cap: "Example"
#| tbl-subcap: 
#|   - "Cars"
#|   - "Pressure"
#| layout-ncol: 2
#| echo: fenced

library(knitr)
kable(head(cars))
kable(head(pressure))

```
![Capture](https://user-images.githubusercontent.com/82851334/171423150-3aa70e5e-e752-445c-82ad-8a901c5e378e.JPG)

**Comment by jjallaire** [comment, marked as answer]
The problem is the space you have left at the top of the chunk (it invalidates all of the options). This works:

![Screen Shot 2022-06-01 at 10 02 54 AM](https://user-images.githubusercontent.com/104391/171423411-b4d5b82c-8936-461b-a3c9-a79e99adcc38.png)

**Comment by sutzig** [comment]
Ah .... thank you.
